cuebot/src/main/java/com/imageworks/spcue/util/FrameSet.java

```diff
@@ -44,6 +48,140 @@ public int index(int idx) {
+        }
+
+        ArrayList<String> resultBuilder = new ArrayList<String>();
+
+        int curr_count = 1;
+        int curr_stride = 0;
+        int new_stride = 0;
+        int curr_start = frames.get(0);
+        int curr_frame = frames.get(0);
+        int last_frame = frames.get(0);
+
+        for (int i = 1; i < frames.size(); i++) {
+            curr_frame = frames.get(i);
+
+            if (curr_stride == 0) {
+                curr_stride = curr_frame - curr_start;
+            }
+            new_stride = curr_frame - last_frame;
+            if (curr_stride == new_stride) {
+                last_frame = curr_frame;
+                curr_count += 1;
+                continue;
+            } else if (curr_count == 2 && curr_stride != 1) {
+                resultBuilder.add(pad(curr_start, zFill));
+                curr_stride = 0;
+                curr_start = last_frame;
+                last_frame = curr_frame;
+                continue;
+            } else {
+                resultBuilder.add(buildFrangePart(curr_start, last_frame, curr_stride, zFill));
+                curr_stride = 0;
+                curr_start = curr_frame;
+                last_frame = curr_frame;
+                curr_count = 1;
+                continue;
+            }
+        }
+        if (curr_count == 2 && curr_stride != 1) {
+            resultBuilder.add(pad(curr_start, zFill));
+            resultBuilder.add(pad(curr_frame, zFill));
+        } else {
+            resultBuilder.add(buildFrangePart(curr_start, curr_frame, curr_stride, zFill));
+        }
+
+        StringBuilder sb = new StringBuilder();
+        l = resultBuilder.size()-1;
+        for (int j = 0; j < l; j++) {
+            sb.append(resultBuilder.get(j)).append(",");
+        }
+        sb.append(resultBuilder.get(l));
+        return sb.toString();
+    }
+
+    /**
+     * Return a sub-FrameSet object starting at startFrame with max chunkSize members
+     * @param idx
+     * @return FrameSet
+     */
+    public String getChunk(int startFrameIndex, int chunkSize) {
+        if (frameList.size() <= startFrameIndex || startFrameIndex < 0) {
+            String sf = String.valueOf(startFrameIndex);
+            String sz = String.valueOf(frameList.size() - 1);
+            throw new SpcueRuntimeException("startFrameIndex " + sf + " is not in range 0-" + sz);
+        }
+        if (chunkSize == 1) {
+            // Chunksize of 1 so the FrameSet is just the startFrame
+            return String.valueOf(frameList.get(startFrameIndex));
+        }
+        int finalFrameIndex = frameList.size() - 1;
+        int endFrameIndex = startFrameIndex + chunkSize;
+        if (endFrameIndex > finalFrameIndex) {
+            // We don't have enough frames, so return the remaining frames.
+            endFrameIndex = finalFrameIndex;
+        }
+
+        ImmutableList<Integer> sl = frameList.subList(startFrameIndex, endFrameIndex);
+
+        return framesToFrameRanges(sl);
```

Comment:
nit: I would just combine this and the line above into a single line -- ditch the variable.